why-semver.qmd

```diff
@@ -0,0 +1,169 @@
+`MAJOR.MINOR.PATCH`. This system is widely used in the software industry
+and is well understood by developers. For instance, with a bug fix,
+`PATCH` is incremented by 1, going from `0.0.0` to `0.0.1`. For `MINOR`,
+you increment by 1 when there is a new feature, so from `0.0.1` to
+`0.1.0`. For `MAJOR`, you increment by 1 when there is a breaking
+change, so from `0.1.0` to `1.0.0`.
+
+::: columns
+::: column
+#### Benefits
+
+-   This is the most widely used versioning system in the software
+    industry, so is very familiar to people.
+-   It is very clear what has changed between versions, so it conveys
+    information about changes well.
+-   Has a large number of tools and packages that support it, combine
+    well with it, or integrate well into development workflows around
+    it. For instance, there are tools that can automatically generate
+    changelogs based on the commit messages in a repository, that can
+    automatically increment the version number based on the changes
+    made, or that
+:::
+
+::: column
+#### Drawbacks
+
+-   It can be very difficult to determine when something is a major,
+    minor, or patch change. Sometimes it is perfectly clear, but often,
+    a change is hard to place in these three categories.
+-   It can be difficult to determine what constitutes a breaking change.
+-   More difficult (but not impossible) to automate, since it can
+    involve a large amount of human judgement.
+:::
+:::
+
+### CalVer
+
+CalVer is a system that is based on dates. Unlike SemVer, there are more
+variations available to use, for instance, `YYYY.MM` or `YY.MM`. Also
+unlike SemVer, there are more often variations that combine features of
+SemVer and CalVer, such as `YYYY.MINOR.PATCH`.
+
+::: columns
+::: column
+#### Benefits
+
+-   Very easy to use and understand, as it is based on dates. Whatever
+    date the software is released on, is the version number.
+-   Works very well for some software projects, especially those with
+    very regular releases. For instance, the most popular Linux
+    distribution, Ubuntu, uses a form of CalVer, that increment every 6
+    months.
+:::
+
+::: column
+#### Drawbacks
+
+-   It doesn't contain any information on what was changed between
+    versions.
+-   Breaking changes are not communicated in the version number.
+-   There are substantially fewer tools and packages to help automate
+    release and development workflows compared to SemVer.
+-   Doesn't work as well for some software projects, especially those
+    that are not released on a regular basis.
+:::
+:::
+
+## Decision outcome
+
+::: content-hidden
+What decision was made, use the form "We decided on CHOICE because of
+REASONS."
+:::
+
+We decided to use Semantic Versioning (SemVar) for versioning our
+software. While it is more difficult to implement and decide on which
+version to increment based on the changes made, because it is so widely
```

Comment:
```suggestion
software. While it can be more difficult to implement as it may require some human judgement to decide on which
version to increment based on the changes made, because it is so widely
```
Is it more difficult to implement in general or bc you need to decide which part of the version number to bump?

Reply:
Difficult because it can be tricky to know whether a new internal function is classified as a "feat" but that is never exposed to the user, though the user might see some performance improvements. Does that warrant an minor update, a patch update? If you strictly go with what SemVer says about minor, it wouldn't fall under an increment, but so what does it fall under? Nothing?